tests/core/test_element_data.py

```diff
@@ -40,3 +40,15 @@ def test_external_id_index_to_iloc(count, expected_missing):
 
     # missing value
     assert idx.to_iloc(["A"]) == expected_missing
+
+
+def test_benchmark_external_id_index_from_iloc(benchmark):
+    N = 1000
+    SIZE = 100
+    idx = ExternalIdIndex(np.arange(N))
+
+    def f():
+        x = np.random.randint(0, N, size=SIZE)
```

Comment:
i think the overhead of this is likely to be similar to `from_iloc` and probably under-estimates the speedup from this PR!

I think a test similar to https://github.com/stellargraph/stellargraph/pull/1267#discussion_r416327081 might be appropriate here. but imo this isn't a blocking concern

Reply:
Err, great catch. The overhead of `randint` is actually dramatically more than `from_iloc`, for 

```python
x = np.random.randint(0, N, size=SIZE)
def f():
    idx.from_iloc(x)
```

The median &pm; iqr is 4.245 &pm; 0.369, i.e. almost 80% of the time is spend doing `randint`. I'm going to go with the simpler form here because the between-run variance doesn't seem too high... I guess we might be able to get away with the simple form in the `StellarGraph` benchmarks too (although we're more likely to have some sort of adaptive algorithms there, e.g. handling very high degree nodes or commonly query nodes).

To validate this, I ran `test_benchmark_external_id_index_from_iloc` several times



```
   Min       Max    Mean  StdDev  Median     IQR  Outliers  OPS (Kops/s)  Rounds  Iterations     
3.9730   76.9699  4.2392  1.6609  4.1041  0.0510  208;1425      235.8910   22239           1
3.7202   52.2460  4.0642  1.1524  4.0079  0.1050  173;619       246.0481   28684           1
3.7029   48.3089  4.1099  1.3056  4.0610  0.0831  145;4052      243.3169   24395           1
3.7120  105.8390  4.1295  2.0666  3.9020  0.1900  275;531       242.1584   27607           1
3.6780   61.1350  4.2788  0.9586  4.3069  0.1958  80;2290       233.7111   25250           1
3.7560   66.0201  4.9839  4.0769  4.1490  0.1751  614;1808      200.6445   16076           1
3.8252   64.5600  4.8152  3.1023  4.2450  0.3690  800;1118      207.6745   25286           1
```

That is a variation of about 10%, which doesn't seem much larger than the variation between runs even without this.

Reply:
woah, so we're looking at a ~50x speedup now?

Reply:
No, 12&times; (~50ms -> ~4ms).